src/cryptonote_core/service_node_voting.h

```diff
@@ -87,13 +108,27 @@ namespace service_nodes
     };
   };
 
-  quorum_vote_t     make_deregister_vote(uint64_t block_height, uint16_t index_in_group, uint16_t worker_index, crypto::public_key const &pub_key, crypto::secret_key const &secret_key);
+  enum struct make_deregister_type { uptime, checkpoint };
+  quorum_vote_t     make_deregister_vote             (uint8_t hf_version,
+                                                      make_deregister_type type,
+                                                      uint64_t block_height,
+                                                      uint16_t index_in_group,
+                                                      uint16_t worker_index,
+                                                      crypto::public_key const &pub_key,
+                                                      crypto::secret_key const &secret_key);
+  bool              verify_tx_deregister             (const cryptonote::tx_extra_service_node_deregister_& deregister,
+                                                      uint64_t latest_height,
+                                                      cryptonote::vote_verification_context& vvc,
+                                                      const service_nodes::testing_quorum &quorum);
+  bool              verify_vote                      (const quorum_vote_t& vote,
+                                                      uint64_t latest_height,
+                                                      cryptonote::vote_verification_context &vvc,
+                                                      const service_nodes::testing_quorum &quorum);
```

Comment:
:nauseated_face: on the excessive whitespace lining up here.

Reply:
alright alright 2v1, majority wins.